imap/monitor.go

```diff
@@ -192,3 +200,45 @@ func checkForNewEmails(im models.IMAP) {
 		log.Debug("No new emails for ", im.Username)
 	}
 }
+
+// returns a slice of gophish rid paramters found in the email HTML, Text, and attachments
+func checkRIDs(em *email.Email) (map[string]int, error) {
+
+	rids := make(map[string]int)
+
+	// Check Text and HTML
+	emailContent := string(em.Text) + string(em.HTML)
+	for _, r := range goPhishRegex.FindAllStringSubmatch(emailContent, -1) {
+		newrid := r[len(r)-1]
+		if _, ok := rids[newrid]; ok {
+			rids[newrid]++
+		} else {
+			rids[newrid] = 1
+		}
+	}
+
+	//Next check each attachment
+	for _, a := range em.Attachments {
+		if a.Header.Get("Content-Type") == "message/rfc822" || (len(a.Filename) > 3 && a.Filename[len(a.Filename)-4:] == ".eml") {
+
+			//Let's decode the email
+			rawBodyStream := bytes.NewReader(a.Content)
+			attachementEmail, err := email.NewEmailFromReader(rawBodyStream)
+			if err != nil {
+				return rids, err
+			}
+
+			emailContent := string(attachementEmail.Text) + string(attachementEmail.HTML)
+			for _, r := range goPhishRegex.FindAllStringSubmatch(emailContent, -1) {
```

Comment:
Since you're repeating this code here, you might consider breaking it out into a function. In fact, I think that `checkRIDs` might actually be a good name for that function since it actually checks for the RIDs in content, and we can rename this one to be `matchEmail` or something - I'm open to suggestions!

Reply:
I did think about this when coding it, and have given it some more thought - it's a little tricky as I'm using a map to maintain the unique list of rids.  I think the easiest solution is to implement the two separate functions as you suggest and pass the rid map so we can update it.

Well, I implemented that, let me know if that's OK or if you have a neater idea :)

Also, I do realise we're not checking attachments of attachments, etc. I can save that for later.